fixbackend/auth/router.py

```diff
@@ -122,14 +129,94 @@ def auth_router(config: Config, google_client: GoogleOAuth2, github_client: Gith
         include_in_schema=False,
     )
 
-    logout_route_name = f"auth:{auth_backend.name}.logout"
-    auth_router = fastapi_users.get_auth_router(auth_backend, requires_verification=True)
-    # remove the logout route, as we have our own
-    auth_router.routes = list(
-        filter(lambda route: not (isinstance(route, Route) and route.name == logout_route_name), auth_router.routes)
-    )
+    @router.post("/mfa/add")
+    async def add_mfa(user: AuthenticatedUser, user_manager: UserManager = Depends(get_user_manager)) -> OTPConfig:
+        if user.is_mfa_active:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="MFA already enabled",
+            )
+        # create a new one-time password
+        user_secret = pyotp.random_base32()
+        totp = pyotp.totp.TOTP(user_secret)
+        # store the secret
+        await user_manager.user_repository.update(user, {"otp_secret": user_secret, "is_mfa_active": False})
```

Comment:
WDYT about depending on `user_repository` here directly, instead of the `user_manager`?

Reply:
yes. will adjust this.